### PR TITLE
Update `cal_step` for new Roman steps

### PR DESCRIPTION
**Pull request by WilliamJamieson**
This PR adds additional steps to the cal_step schema

**Checklist**
- [ ] Schema changes discussed at RAD Review Board meeting
- [x] Added entry in `CHANGES.rst` under the corresponding subsection
- [x] Updated relevant roman_datamodels utilities and tests
- [x] Passed romancal regression testing on Jenkins / PLWishMaster. Link: https://plwishmaster.stsci.edu:8081/job/RT/job/Roman-Developers-Pull-Requests/262//

### CODE REVIEW

**Review by kdupriestsci — APPROVED** [review]

**Comment by WilliamJamieson** [comment]
@nden and @ddavis-stsci can you take a look at the new `cal_step` names I added and let me know if those are what we want to go with. Namely, Outlier Detection and Tweakreg did not have cal_step entries, so I guessed at names.

**Review by ddavis-stsci — COMMENTED** [review]
Looks good. Can you also add skymatch?

**Comment by WilliamJamieson** [comment]
This PR will break `roman_datamodels`, but spacetelescope/roman_datamodels#228 will fix those issues.

File: CHANGES.rst [hunk 1]
```diff
@@ -1,3 +1,9 @@
+0.17.0 (unreleased)
+-------------------
+
+- Update the steps listed in ``cal_steps`` to reflect the currently implemented steps.
+  The new additions are ``outlier_detection``, ``refpix``, ``sky_match``, and ``tweak_reg``. [#282]
```
Comment (nden) [inline CHANGES.rst:5]: `tweak_reg` --> `tweakreg`

File: src/rad/resources/schemas/cal_step-1.0.0.yaml [hunk 2]
```diff
@@ -69,14 +69,42 @@ properties:
...
       destination: [ScienceRefData.saturation, GuideWindow.saturation]
-propertyOrder: [assign_wcs, flat_field, dark, dq_init, jump, linearity, photom, source_detection, ramp_fit, saturation]
+  outlier_detection:
+    title: outlier_detection
+    type: string
+    enum: ['N/A', 'COMPLETE', 'SKIPPED', 'INCOMPLETE']
+    archive_catalog:
+      datatype: nvarchar(15)
+      destination: [ScienceRefData.outlier_detection, GuideWindow.outlier_detection]
+  tweak_reg:
```
Comment (nden) [inline src/rad/resources/schemas/cal_step-1.0.0.yaml:93]:
```suggestion
  tweakreg:
```

File: src/rad/resources/schemas/cal_step-1.0.0.yaml [hunk 3]
```diff
@@ -69,14 +69,42 @@ properties:
...
+    title: Tweakreg step
+    type: string
+    enum: ['N/A', 'COMPLETE', 'SKIPPED', 'INCOMPLETE']
+    archive_catalog:
+      datatype: nvarchar(15)
+      destination: [ScienceRefData.tweak_reg, GuideWindow.tweak_reg]
+  sky_match:
```
Comment (nden) [inline src/rad/resources/schemas/cal_step-1.0.0.yaml:100]:
```suggestion
  skymatch:
```

File: src/rad/resources/schemas/cal_step-1.0.0.yaml [hunk 4]
```diff
@@ -69,14 +69,42 @@ properties:
...
+    title: Sky Match step
+    type: string
+    enum: ['N/A', 'COMPLETE', 'SKIPPED', 'INCOMPLETE']
+    archive_catalog:
+      datatype: nvarchar(15)
+      destination: [ScienceRefData.sky_match, GuideWindow.sky_match]
+propertyOrder: [assign_wcs, flat_field, dark, dq_init, jump, linearity, photom, source_detection, outlier_detection, ramp_fit, refpix, saturation, sky_match, tweak_reg]
```
Comment (nden) [inline src/rad/resources/schemas/cal_step-1.0.0.yaml:107]:
```suggestion
propertyOrder: [assign_wcs, flat_field, dark, dq_init, jump, linearity, photom, source_detection, outlier_detection, ramp_fit, refpix, saturation, skymatch, tweakreg]
```

**Comment by nden** [comment]
Hm, it's likely there's no need to match the `cal_step` entry with the name of the step in which case, ignore my suggestions.

**Review by PaulHuwe — APPROVED** [review]
LGTM - if this will affect regression tests, please wait to merge until we are ready for a mass update to files.

**Comment by WilliamJamieson** [comment]
No this does not change anything for the regressions. As noted it does not set the values as required. That can be done prior to the next generation of registration test data.

**Comment by PaulHuwe** [comment]
Okay, then I am good with merging.

**Comment by WilliamJamieson** [comment]
> Hm, it's likely there's no need to match the `cal_step` entry with the name of the step in which case, ignore my suggestions.

I like your suggestions better. So I kept them, right before the rebuild of all the files I think we should take a moment to review all these step names and sync them across romancal and rad. That way its all uniform.

**Pull request merged**